src/object.c

```diff
@@ -140,7 +140,7 @@ robj *createRawStringObject(const char *ptr, size_t len) {
 }
 
 /* Get beginning of embedded data, which may contain expire, key, and/or value. Embedded data flags must be accurate when called. */
-static unsigned char *objectEmbeddedData(const robj *o) {
+__attribute__((pure)) static unsigned char *objectEmbeddedData(const robj *o) {
```

Comment:
The pure/const changes look correct.  Is there a reason that the implementations include the attribute on the left, and the forward declarations include the attribute on the right?